packages/export-handler/src/index.ts

```diff
@@ -81,103 +100,132 @@ export const exporter = Sentry.GCPFunction.wrapHttpFunction(
+      const archive = archiver('zip', {
+        zlib: { level: 9 }, // Compression level
+      })
+
+      // Handle any archiver errors
+      archive.on('error', (err) => {
+        throw err
+      })
+
+      // Pipe the archiver output to the PassThrough stream
+      archive.pipe(passthroughStream)
 
       // fetch data from the database
       const omnivore = new Omnivore({
         apiKey: claims.token,
       })
 
+      const batchSize = 20
       let cursor = 0
       let hasNext = false
       do {
         const response = await omnivore.items.search({
-          first: 100,
+          first: batchSize,
           after: cursor,
-          includeContent: false,
+          includeContent: true,
+          query: 'in:all',
         })
 
         const items = response.edges.map((edge) => edge.node)
-        cursor = response.pageInfo.endCursor
-          ? parseInt(response.pageInfo.endCursor)
-          : 0
-        hasNext = response.pageInfo.hasNextPage
 
+        const size = items.length
         // write data to the csv file
-        if (items.length > 0) {
-          // write the list of urls, state and labels to the stream
-          items.forEach((item) =>
-            stringifier.write({
-              id: item.id,
-              title: item.title,
-              description: item.description,
-              labels: item.labels?.map((label) => label.name).join(','),
-              author: item.author,
-              site_name: item.siteName,
-              original_url: item.originalArticleUrl,
-              slug: item.slug,
-              updated_at: item.updatedAt,
-              saved_at: item.savedAt,
-              type: item.pageType,
-              published_at: item.publishedAt,
-              url: item.url,
-              thumbnail: item.image,
-              read_at: item.readAt,
-              word_count: item.wordsCount,
-              reading_progress_percent: item.readingProgressPercent,
-              archived_at: item.archivedAt,
-            })
-          )
-
-          // sleep for 1 second to avoid rate limiting
-          await new Promise((resolve) => setTimeout(resolve, 1000))
+        if (size > 0) {
+          // Add the metadata.json file to the root of the zip
+          const metadata = items.map((item) => ({
+            id: item.id,
+            slug: item.slug,
+            title: item.title,
+            description: item.description,
+            author: item.author,
+            url: item.originalArticleUrl,
+            state: item.isArchived ? 'archived' : 'active',
+            readingProgress: item.readingProgressPercent,
+            thumbnail: item.image,
+            labels: item.labels?.map((label) => label.name),
+            savedAt: item.savedAt,
+            updatedAt: item.updatedAt,
+            publishedAt: item.publishedAt,
+          }))
+
+          archive.append(JSON.stringify(metadata, null, 2), {
+            name: `metadata_${cursor}_to_${cursor + size}.json`,
```

Comment:
@jacksonh Here I actually export paginated `metadata` into multiple JSON files to avoid having a very large JSON file